register/index.js

```diff
@@ -0,0 +1,114 @@
+"use strict";
+
+const { Client } = require('pg');
+const bcrypt = require('bcrypt');
+
+const client = new Client({
+  user: 'joseph',
+  host: 'localhost',
+  database: 'hackers',
+  password: 'password',
+  post: 5432
+});
+
+client.connect().then(() => console.log('connected to database...')).catch(e => console.log(e));
+
+/*
+    https://www.npmjs.com/package/bcrypt
+    Note: Since bcrypt.hash returns a promise (when the callback isn't specified), we need to resolve that 
+    before we return the hashed password or we'll just get a Promise { <pending> }. 
+*/
+const createHashedPw = async (password) => {
+  const saltRounds = 10;
+  const hashedPassword = await new Promise((resolve, reject) => {
+    bcrypt.hash(password, saltRounds, (err, hash) => {
+      if (err) reject(err);
+      else resolve(hash);
+    });
+  });
+  return hashedPassword;
+};
+
+/*
+    Helper function to create the correct SQL command.
+*/
+const sqlCommandGenerator = async (req) => {
+    // First hash the password
+    const hashedPassword = await createHashedPw(req.body.attendeePassword);
+    // this is the sql command (possibly use an ORM to simplify?)
+    let text = "INSERT INTO attendees"+
+    "(name,email,password,age,school,year_of_graduation,"+
+    "ucsc_student,ucsc_college_affil,major,linkedin_url,"+
+    "github_url,first_hackathon,first_cruzhacks,particate_question,"+
+    "humanity_goals_question,cruzhacks2020_question,has_a_team,team_members,"+
+    "team_member_emails,dietary_restrictions,housing_accomadation,"+
+    "bussing_accomadation,place_to_park) VALUES($1, $2, $3, $4, $5"+
+    ", $6, $7, $8, $9, $10, $11, $12, $13, $14, $15, $16, $17, $18, $19, $20, $21,"+
+    " $22, $23) RETURNING *";
+
+    let valuesObj = {
```

Comment:
Good start inserting database fields. We're dealing with multiple types of applications however, so it may be ideal to setup this object to describe base fields instead; only instantiating fields that are shared across all applications. From there, we can use a switch statement to select our application type from the http request.

Reply:
Hey @mocklee, could you help me on this one? Did you want like something with the base values held in a class (for example):

class User {
       constructor(username, password, email)
       ...
       ...
}

class Hacker extends User {
       ...
}

class Mentor extends User {

}

Reply:
Switch statements might work better; classes would make our code longer. To 'extend' the base fields, use switch cases to add the appropriate fields to the request: https://www.w3schools.com/js/js_switch.asp

Reply:
We would set switch cases to be a match for whatever route the request came from, like req.path.includes('volunteer')

Reply:
I wasn't sure if we could make a microservice direct towards 1 url? 
For example: /api/register would be our main url, and we have /api/register/hacker, /api/register/volunteer, /api/register/mentor that call the service at /api/register? I thought a microservice would only act on 1 url (/api/register/hacker and the rest of them would be a separate service).

Reply:
I think I get it now, sorta like something from this article?

https://blogs.msdn.microsoft.com/petsablog/2016/06/06/serverless-azure-architecture-node-js-azure-functions-an-api-management/